multidimensional_urlencode/urlencoder.py

```diff
@@ -23,7 +26,7 @@ def flatten(d):
         return [[d]]
 
     returned = []
-    for key, value in d.items():
+    for key, value in sorted(list(d.items())):
```

Comment:
Actually, in both cases the order is undefined if I recall correctly. It seems a bit heavy handed to this sorting inside the library code, because that might be prohibitively expensive. What about just doing this formatting in tests? Or use `pprint`?